Update metadata services.

hgreenlee opened the pull request:
Add ability to generate metadata for multiple files and other features.

cjbackhouse commented on sbncode/Metadata/FileCatalogMetadataSBN.h:
```diff
@@ -6,11 +6,18 @@
+//           ProjectSoftware - Project software information.
+//           ProductionName  - Production campaign name.
+//           ProductionType  - Production campaign type.
+//           Merge           - Merge flag.
```
I found this pretty cryptic. Can you explain further (in this comment?) what the merge fcl parameter represents, and what the merge parameters in the metadata represent?

hgreenlee replied:
At this moment parameters merge.merge and merge.merged are just dumb sam parameters that don't do anything.  The intended semantic meaning of parameter merge.merge is that it flags a file that wants to be merged.  Parameter merge.merged flags a file that has been merged.  In MicroBooNE, parameter merge.merge causes files to flow from batch jobs into a special merging FTS dropbox which stores unmerged files temporarily on disk.  Both parameters are used to bookkeep the subsequent merging process.

cjbackhouse commented on sbncode/Metadata/FileCatalogMetadataSBN.h:
```diff
@@ -66,6 +74,8 @@ namespace util {
     std::string fProductionName; //Production parameter, do not use if not running a production
     std::string fProductionType; //Production parameter, do not use if not running a production
+    int fMerge;
+    std::vector<std::string> fParameters;
```
Can we use `vector<pair<string, string>>` or `map<string, string>` please?

hgreenlee replied:
The reason I had it as vector<string> is because that is a supported type by fcl.  I'm OK with switching to a pairwise type.

cjbackhouse commented on sbncode/Metadata/FileCatalogMetadataSBN_service.cc:
```diff
@@ -27,7 +28,15 @@ FileCatalogMetadataSBN(fhicl::ParameterSet const& pset, art::ActivityRegistry& r
   fProductionName = pset.get<std::string>("ProductionName","");  //Leave as default value if not running a production   
   fProductionType = pset.get<std::string>("ProductionType",""); //Leave as default value if not running a production
+  fMerge = pset.get<int>("Merge", -1);
+  fParameters = pset.get<std::vector<std::string> >("Parameters", std::vector<std::string>());
```
And have parameters represented in the fcl file as nested vectors `[[key1, value1], [key2, value2], ...`

hgreenlee replied:
Or a fcl table.

cjbackhouse reviewed (changes requested):
Looks OK, except for a couple of issues I flagged.

hgreenlee commented:
Changed fcl parameter "Parameters" in FileCatalogMetadataSBN from plain vector-of-strings to vector-of-pairs-of-strings.
Added fcl parameter "Experiment," which was missing.

hgreenlee commented:
Updated to add support for sam parameter mc.pot.

jzennamo commented:
This looks great! I didn't have any comments, approved on my side

FNALbuild commented:
CI build for LArSoft on slf7 for e20:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/11899&builds=)

FNALbuild commented:
CI build for LArSoft on slf7 for c7:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/11898&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c7:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/11898&builds=)

FNALbuild commented:
CI build for SBND on slf7 for c7:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/6738&builds=)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/11898&builds=)

FNALbuild commented:
CI build for ICARUS on slf7 for c7:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2172&builds=)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/11898&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase build ICARUS on slf7 for c7:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2172&builds=)

:rotating_light: For more details about the failed phase, check the [build ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/2172&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=build&buildtype=slf7%20c7:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/11898&builds=)

FNALbuild commented:
:warning: CI build for SBND Warning at phase ci_tests SBND on slf7 for c7:prof - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/6738&builds=)

:rotating_light: For more details about the warning phase, check the [ci_tests SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/6738&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=ci_tests&buildtype=slf7%20c7:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/11898&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e20:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/11899&builds=)

FNALbuild commented:
CI build for ICARUS on slf7 for e20:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2173&builds=)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/11899&builds=)

FNALbuild commented:
CI build for SBND on slf7 for e20:prof is in progress -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/6739&builds=)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/11899&builds=)

FNALbuild commented:
:warning: CI build for SBND Warning at phase ci_tests SBND on slf7 for e20:prof - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/6739&builds=)

:rotating_light: For more details about the warning phase, check the [ci_tests SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/6739&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=ci_tests&buildtype=slf7%20e20:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/11899&builds=)

FNALbuild commented:
:warning: CI build for ICARUS Warning at phase ci_tests ICARUS on slf7 for e20:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/2173&builds=)

:rotating_light: For more details about the warning phase, check the [ci_tests ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/2173&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=ci_tests&buildtype=slf7%20e20:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/11899&builds=)
